test/e2e/framework.go

```diff
@@ -1152,3 +1175,70 @@ func (data *TestData) mutateAntreaConfigMap(mutatingFunc func(data map[string]st
 	}
 	return nil
 }
+
+// gracefulExitAntreaController copies the Antrea controller binary coverage data file out before terminating the Pod
+func (data *TestData) gracefulExitAntreaController(covDir string) error {
+	antreaController, err := data.getAntreaController()
+	if err != nil {
+		return fmt.Errorf("Error when getting antrea-controller Pod: %v", err)
+	}
+	podName := antreaController.Name
+	cmds := []string{"pgrep", "-o", antreaControllerCovBinary, "-P", "1"}
+	stdout, stderr, err := data.runCommandFromPod(antreaNamespace, podName, "antrea-controller", cmds)
+	if err != nil {
+		return fmt.Errorf("error when getting pid of antreaContollerCovBinary: <%v>, err: <%v>", stderr, err)
+	}
+	cmds = []string{"kill", "-SIGINT", strings.TrimSpace(stdout)}
+
+	_, stderr, err = data.runCommandFromPod(antreaNamespace, podName, "antrea-controller", cmds)
+	if err != nil {
+		return fmt.Errorf("error when sending SIGINT signal to antreaContollerCovBinary: <%v>, err: <%v>", stderr, err)
+	}
+	// a filepath-friendly timestamp format.
+	const timeFormat = "Jan02-15-04-05"
+	timeStamp := time.Now().Format(timeFormat)
+	rc, _, stderr, err := RunCommandOnNode(clusterInfo.masterNodeName, fmt.Sprintf("kubectl cp %s/%s:%s %s/antrea-controller-%s.out", antreaNamespace, podName, antreaControllerCovFile, covDir, timeStamp))
+	if err != nil || rc != 0 {
+		return fmt.Errorf("error when graceful exit Antrea: copy %s from pod %s, stderr: %s, error:%v", antreaControllerCovFile, podName, stderr, err)
+	}
+	return nil
+}
+
+// gracefulExitAntreaAgent copies the Antrea agent binary coverage data file out before terminating the Pod
+func (data *TestData) gracefulExitAntreaAgent(covDir string, nodeName string) error {
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=antrea,component=antrea-agent",
+	}
+	if nodeName != "all" {
+		listOptions.FieldSelector = fmt.Sprintf("spec.nodeName=%s", nodeName)
+	}
+
+	pods, err := data.clientset.CoreV1().Pods(antreaNamespace).List(context.TODO(), listOptions)
+	if err != nil {
+		return fmt.Errorf("failed to list antrea-agent pods: %v", err)
+	}
+	for _, pod := range pods.Items {
+		podName := pod.Name
+		cmds := []string{"pgrep", "-o", antreaAgentCovBinary, "-P", "1"}
+		stdout, stderr, err := data.runCommandFromPod(antreaNamespace, podName, "antrea-agent", cmds)
+		if err != nil {
+			return fmt.Errorf("error when getting pid of antreaAgentCovBinary: <%v>, err: <%v>", stderr, err)
+		}
+		fmt.Printf("pid: %s\n", stdout)
+		cmds = []string{"kill", "-SIGINT", strings.TrimSpace(stdout)}
+		stdout, stderr, err = data.runCommandFromPod(antreaNamespace, podName, "antrea-agent", cmds)
```

Comment:
stdout never used, please address

Reply:
Good catch, stdout was added when debugging, now changed to _.